core/trino-main/src/main/java/io/trino/execution/executor/TaskExecutor.java

```diff
@@ -307,6 +314,16 @@ private void doRemoveTask(TaskHandle taskHandle)
         log.debug("Task finished or failed %s", taskHandle.getTaskId());
     }
 
+    private synchronized void recordAllSplitQueueSize()
```

Comment:
how expensive is updating these. Could it cause perf (lock congestion issues)?

Reply:
You can look here: https://github.com/trinodb/trino/pull/13453#issuecomment-1217718325
I've generated JFR profile and I do not see that there is "hot" lock there.